src/api/emit.js

```diff
@@ -10,11 +10,16 @@ const CustomEvent = ((Event) => {
 })(window.CustomEvent);
 
 function createCustomEvent(name, opts = {}) {
-  if (CustomEvent) {
-    return new CustomEvent(name, opts);
+  let e;
+  if (Event) {
+    e = new Event(name, opts);
+    if (opts.detail) {
```

Comment:
Might be okay to leave this out, or do `'detail' in opts` so that falsy values can still be passed.